studio/LonaStudio/Workspace/FileNavigator.swift

```diff
@@ -185,6 +185,36 @@ class FileNavigator: NSBox {
                 menu.addItem(NSMenuItem.separator())
             }
 
+            if NSURL(fileURLWithPath: path).pathExtension == "component"{
+                menu.addItem(NSMenuItem(title: "Duplicate As...", onClick: {
+                    var saveURL: String
+                    let dialog = NSSavePanel()
+
+                    dialog.title                   = "Save .component file"
+                    dialog.showsResizeIndicator    = true
+                    dialog.showsHiddenFiles        = false
+                    dialog.canCreateDirectories    = true
+                    dialog.allowedFileTypes        = ["component"]
+
+                    // User canceled the save. Don't swap out the document.
+                    if dialog.runModal() != NSApplication.ModalResponse.OK {
+                        return
+                    }
+
+                    guard let url = dialog.url else { return }
+                    saveURL = url.path
+                    do {
+                        try FileManager.default.copyItem(atPath: path, toPath: saveURL)
+                        self.onAction?(saveURL)
```

Comment:
If we call `self.onAction?(saveURL)` in the `do ... catch` block, it will swallow errors thrown when opening the file, which probably isn't what we want.

Let's move this call to after the `do ... catch` block. We can add a `return` to the end of the `catch`.